app/controllers/api/migrations.php

```diff
@@ -0,0 +1,554 @@
+    });
+
+App::post('/v1/migrations/:migrationId')
+    ->groups(['api', 'migrations'])
+    ->desc('Retry Migration')
+    ->label('scope', 'migrations.write')
+    ->label('event', 'migrations.[migrationId].retry')
+    ->label('audits.event', 'migration.retry')
+    ->label('audits.resource', 'migrations/{request.migrationId}')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'retry')
+    ->label('sdk.description', '/docs/references/migrations/retry-migration.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION)
+    ->param('migrationId', '', new UID(), 'Migration unique ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (string $migrationId, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventInstance) {
+        $migration = $dbForProject->getDocument('migrations', $migrationId);
+
+        if ($migration->isEmpty()) {
+            throw new Exception(Exception::MIGRATION_NOT_FOUND);
+        }
+
+        // if ($migration->getAttribute('status') !== 'failed') {
+        //     throw new Exception(Exception::MIGRATION_IN_PROGRESS, 'Migration not failed');
+        // }
+
+        $migration
+            ->setAttribute('status', 'pending')
+            ->setAttribute('dateUpdated', \time());
+
+        // Trigger Migration
+        $event = new Migration();
+        $event
+            ->setMigration($migration)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response->noContent();
+    });
+
+App::delete('/v1/migrations/:migrationId')
+    ->groups(['api', 'migrations'])
+    ->desc('Delete Migration')
+    ->label('scope', 'migrations.write')
+    ->label('event', 'migrations.[migrationId].delete')
+    ->label('audits.event', 'migrationId.delete')
+    ->label('audits.resource', 'migrations/{request.migrationId}')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'delete')
+    ->label('sdk.description', '/docs/references/functions/delete-migration.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('migrationId', '', new UID(), 'Migration ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('deletes')
+    ->inject('events')
+    ->action(function (string $migrationId, Response $response, Database $dbForProject, Delete $deletes, Event $events) {
+
+        $migration = $dbForProject->getDocument('migrations', $migrationId);
+
+        if ($migration->isEmpty()) {
+            throw new Exception(Exception::MIGRATION_NOT_FOUND, 'Migration not found', 404);
```

Comment:
```suggestion
            throw new Exception(Exception::MIGRATION_NOT_FOUND);
```
Check this for all exceptions.. If the type MIGRATION_NOT_FOUND has a message, we dont need to use the same message and code again unless we want to over write it